File: whatsapp/static/tests/mock_server/mock_models/mail_thread.js
import { mailModels } from "@mail/../tests/mail_test_helpers";

export class MailThread extends mailModels.MailThread {
    /**
     * @override
     * @type {typeof mailModels.MailThread["prototype"]["_thread_to_store"]}
     */
    _thread_to_store(ids, store, request_list) {
        /** @type {import("mock_models").WhatsAppTemplate} */
        const WhatsAppTemplate = this.env["whatsapp.template"];
        super._thread_to_store(...arguments);
        store.add("mail.thread", {
            id: ids[0],
            model: this._name,
            canSendWhatsapp:
                WhatsAppTemplate.search_count([
                    ["model", "=", this._name],
                    ["status", "=", "approved"],
                ]) > 0,
        });
    }
}

File: whatsapp/models/mail_thread.py
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models
from odoo.addons.mail.tools.discuss import Store


class MailThread(models.AbstractModel):
    _inherit = 'mail.thread'

    def _thread_to_store(self, store: Store, **kwargs):
        super()._thread_to_store(store, **kwargs)
        store.add(
            "Thread",
            {
                "canSendWhatsapp": self.env['whatsapp.template']._can_use_whatsapp(self._name),
                "id": self.id,
                "model": self._name,
            },
        )
